src/ImageSharp/Processing/Processors/Normalization/GlobalHistogramEqualizationProcessor{TPixel}.cs

```diff
@@ -123,7 +123,8 @@ public void Invoke(int y)
                     // TODO: We should bulk convert here.
                     var vector = Unsafe.Add(ref pixelBase, x).ToVector4();
                     int luminance = ImageMaths.GetBT709Luminance(ref vector, levels);
-                    Unsafe.Add(ref histogramBase, luminance)++;
+                    ref int histogramAtLuminance = ref MemoryMarshal.GetReference(this.histogramBuffer.Slice(luminance));
+                    Interlocked.Increment(ref histogramAtLuminance);
```

Comment:
Couldn't this have been

```c#
Interlocked.Increment(ref Unsafe.Add(ref histogramBase, luminance))
```

Reply:
yes right, dont know why i did it so over complicated

Reply:
not sure why net472 tests are failing now. Its not the histogram tests:

`Unhandled Exception: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
   at ImageMagick.PixelCollection.NativeMethods.X86.PixelCollection_ToByteArray(IntPtr Instance, UIntPtr x, UIntPtr y, UIntPtr width, UIntPtr height, IntPtr mapping, IntPtr& exception)
   at ImageMagick.PixelCollection.NativePixelCollection.ToByteArray(Int32 x, Int32 y, Int32 width, Int32 height, String mapping)
  ...`

Reply:
It's an intermittent issue with ImageMagick. Maybe caused by OOM but I'm not sure. I just reset when I see it.